# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gtest/TestAudioPacketizer.cpp
# Commit: c504995adbb4
# Full Hash: c504995adbb459ffb38efbc80540b0e2dcf43ad4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Add methods to inspect buffered amount and to clear AudioPacketizer. r=padenot
#   
#   This patch also constifies members where applicable, and adds a unittest for the
#   new methods.
#   
# ==============================================================================

diff -r 605d00735b78 -r c504995adbb4 dom/media/gtest/TestAudioPacketizer.cpp
--- a/dom/media/gtest/TestAudioPacketizer.cpp	Thu Nov 05 16:42:20 2020 +0000
+++ b/dom/media/gtest/TestAudioPacketizer.cpp	Thu Nov 05 16:42:28 2020 +0000
@@ -136,7 +136,7 @@
         ap.Input(b.Get(), 128);
         while (ap.PacketsAvailable()) {
           std::unique_ptr<int16_t[]> packet(ap.Output());
-          for (uint32_t k = 0; k < ap.PacketSize(); k++) {
+          for (uint32_t k = 0; k < ap.mPacketSize; k++) {
             for (int32_t c = 0; c < channels; c++) {
               ASSERT_TRUE(packet[k * channels + c] ==
                           static_cast<int16_t>(((2 << 14) * sine(outPhase))));
@@ -146,5 +146,18 @@
         }
       }
     }
+    // Test that clearing the packetizer empties it and starts returning zeros.
+    {
+      AudioPacketizer<int16_t, int16_t> ap(441, channels);
+      AutoBuffer<int16_t> b(440 * channels);
+      Sequence(b.Get(), 440 * channels);
+      ap.Input(b.Get(), 440);
+      EXPECT_EQ(ap.FramesAvailable(), 440U);
+      ap.Clear();
+      EXPECT_EQ(ap.FramesAvailable(), 0U);
+      EXPECT_TRUE(ap.Empty());
+      std::unique_ptr<int16_t[]> out(ap.Output());
+      Zero(std::move(out), 441);
+    }
   }
 }